benchmark/csharp/Tpch/VectorFunctions.cs

```diff
@@ -3,49 +3,30 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
-using Apache.Arrow;
+using Microsoft.Data.Analysis;
 
 namespace Tpch
 {
     internal static class VectorFunctions
     {
-        internal static DoubleArray ComputeTotal(DoubleArray price, DoubleArray discount, DoubleArray tax)
+        internal static PrimitiveDataFrameColumn<double> ComputeTotal(PrimitiveDataFrameColumn<double> price, PrimitiveDataFrameColumn<double> discount, PrimitiveDataFrameColumn<double> tax)
         {
             if ((price.Length != discount.Length) || (price.Length != tax.Length))
             {
                 throw new ArgumentException("Arrays need to be the same length");
             }
 
-            int length = price.Length;
-            var builder = new DoubleArray.Builder().Reserve(length);
-            ReadOnlySpan<double> prices = price.Values;
-            ReadOnlySpan<double> discounts = discount.Values;
-            ReadOnlySpan<double> taxes = tax.Values;
-            for (int i = 0; i < length; ++i)
-            {
-                builder.Append(prices[i] * (1 - discounts[i]) * (1 + taxes[i]));
-            }
-
-            return builder.Build();
+            return (PrimitiveDataFrameColumn<double>)(price * (1 - discount) * (1 + tax));
```

Comment:
Do we need to cast here? Doing match on a double column should return a double column, right?

Reply:
Yea, we need a cast unfortunately. The math operators are defined on the base `DataFrameColumn` class and the operator overloads return `DataFrameColumn`. We have https://github.com/dotnet/corefxlab/issues/2822 filed to improve this. When I think about it now, maybe they should've been defined in `PrimitiveDataFrameColumn<T>` instead. No other column can do math anyway. We could roll with this for now and modify it once we fix 2822